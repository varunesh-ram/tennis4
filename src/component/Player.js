import React from 'react';
import PropTypes from 'prop-types';

export default class Player extends React.Component {
    render() {
        return (
            <div>
                <h5>{this.props.name}</h5>
            </div>
        );
    }
}

Player.propTypes = {
    name: PropTypes.string.isRequired
}