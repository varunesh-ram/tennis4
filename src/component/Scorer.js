import React from 'react';
import { Constants } from '../constants/Constants';

export default class Scorer extends React.Component {
    render() {
        return (
            <div>
                <h5>{Constants.SCORER_HEADER}</h5>
                <label>{Constants.INITIAL_SCORE}</label>
            </div>
        );
    }
}