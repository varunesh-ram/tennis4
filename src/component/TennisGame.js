import React from 'react';
import Player from './Player';
import Scorer from './Scorer';
import { Constants } from '../constants/Constants';

export default class TennisGame extends React.Component {
    render() {
        return (
            <div>
                <div className="playerContainer">
                    <div className="leftContainer">
                        <Player name={Constants.PLAYER1_NAME} />
                    </div>
                    <Player name={Constants.PLAYER2_NAME} />
                </div>
                <Scorer />
            </div>);
    }
}