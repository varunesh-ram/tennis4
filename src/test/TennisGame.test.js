import React from 'react';
import { mount } from 'enzyme';
import TennisGame from "../component/TennisGame";
import Player from "../component/Player";
import Scorer from "../component/Scorer";

describe(("<TennisGame/> component"), () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(< TennisGame />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have two players and one scorer", () => {
    expect(wrapper.find(Player).length).toBe(2);
    expect(wrapper.find(Scorer).length).toBe(1);
  });
});
describe(("<TennisGame/> Game functionality"), () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(< TennisGame />);
  });

  it("On Game Start, Score Should be Love All", () => {
    expect(wrapper.find(Scorer).find("label").text()).toEqual("Love all");
  });
});