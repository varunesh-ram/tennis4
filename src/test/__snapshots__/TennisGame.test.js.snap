// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TennisGame/> component should render correctly 1`] = `ReactWrapper {}`;
